sdk/go/common/workspace/templates.go

```diff
@@ -528,6 +528,10 @@ func CopyTemplateFiles(
 		func(entry os.DirEntry, source string, dest string) error {
 			if entry.IsDir() {
 				// Create the destination directory.
+				if force {
+					// MkdirAll will overwrite existing directories
+					return os.MkdirAll(dest, 0o700)
```

Comment:
This fixes part of the original issue, but I don't believe this fixes it completely.

Say the source has a directory `test` with a file `test.txt` in it. We can get the error mentioned in the original issue both by having a destination directory that has a *directory* `test` in it (in which case the code here will fix the issue), or by having a destination directory that has a *file* `test` in it (in which case we will still fail here).  
 
I think the following would work instead:

```suggestion
					info, err := os.Stat(dest)
					if err != nil {
						return err
					}
					if !info.IsDir() {
						os.Remove(dest)
					}
					// MkdirAll will not error out if dest is a directory that already exists
					return os.MkdirAll(dest, 0o700)
```

Reply:
Thank you, good point on the edge case! Looks like we should also handle the case where the dest has directory and we are trying to overwrite it with a file of the same name (this will currently fail and isn't handled by your suggestion since entry is not a dir) - also your suggestion fails on the general case of copying without overwrite - os.Stat errors if the file/dir is not found, and in that case we want to create the directory, not return the error - I added unit tests for these two cases and updated the logic, let me know what you think.

Reply:
Thanks, I think your additions are great here!